### PR TITLE
Switch to using libnetwork's resolvconf package

### LINKED ISSUE #1533: nameservers in container's /etc/resolv.conf should be reachable
When launching `podman run` without `--dns` option, host's `/etc/resolv.conf` is used for the container.

This is a problem when host's `/etc/resolv.conf` only contains nameservers not reachable from the container, e.g. 127.0.0.1.  

Docker ignores  127.0.0.1 in host's `/etc/resolv.conf`. When there’s no other nameserver, it adds 8.8.8.8 and 8.8.4.4.

**Is this a BUG REPORT or FEATURE REQUEST?**:
**kind bug**

### PR DESCRIPTION
**Pull request by mheon**
Libnetwork provides a well-tested package for generating resolv.conf from the host's that has some features our current implementation does not. Swap to using their code and remove our built-in implementation.

Fixes #1533

### CODE REVIEW

**Comment by mheon** [comment]
Alright, they're still on the old Sirupsen package in the tag I used, but not master.

Let me bump to master...

**Comment by mheon** [comment]
Alright, the vendoring situation is getting absurd. Going to move the code into a pkg/ dir in our repo.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by mheon** [comment]
Failing cloud podman is due to Podman 0.9.3 making its way into the crio-tested repo, so we're pulling the version with the bug we fixed in 0.9.3.1

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
rebase

**Comment by vrothberg** [comment]
bot, retest this please

File: vendor.conf [hunk 1]
```diff
@@ -97,3 +97,4 @@ github.com/openshift/imagebuilder master
 github.com/ulikunitz/xz v0.5.4
 github.com/mailru/easyjson 03f2033d19d5860aef995fe360ac7d395cd8ce65
 github.com/coreos/go-iptables 25d087f3cffd9aedc0c2b7eff25f23cbf3c20fe1
+github.com/docker/libnetwork 20461b8539336a4b5fcf551a86dd24ebae211984
```
Comment (vrothberg) [inline vendor.conf]: I think this can be removed given the code went into a dedicated package.

Comment (mheon) [reply vendor.conf]: Good call, fixed

**Comment by mheon** [comment]
@rhatdan @baude @TomSweeneyRedHat @vrothberg PTAL

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
LGTM

**Comment by rhatdan** [comment]
Lots of failures?

**Comment by mheon** [comment]
Network flakes. Always the network.

**Comment by mheon** [comment]
bot, retest this please

**Comment by vrothberg** [comment]
@mheon, looks like some integration tests are unhappy.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, I suspect some of the userns tests may need some tweaking though:

```
/go/src/github.com/containers/libpod/test/e2e/run_userns_test.go:11
  podman uidmapping and gidmapping --net=host [It]
  /go/src/github.com/containers/libpod/test/e2e/run_userns_test.go:68

  Expected
      <int>: 127
  to equal
      <int>: 0
```

**Comment by mheon** [comment]
Pushed again, integration tests should hopefully be fixed.

**Comment by mheon** [comment]
All green now

**Pull request merged**